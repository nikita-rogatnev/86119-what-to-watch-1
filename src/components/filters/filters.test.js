import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import Filters from "./filters";

import mockFilters from "../../mocks/mock-filters";

Enzyme.configure({adapter: new Adapter()});

it(`Filters renders correctly`, () => {
  const tree = shallow(<Filters
    filters={mockFilters}
    currentFilter={`All genres`}
    changeCurrentFilter={() => {
    }}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});

