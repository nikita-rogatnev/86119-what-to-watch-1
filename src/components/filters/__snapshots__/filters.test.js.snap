// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    key="All genres"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=All genres"
    >
      All genres
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Adventure"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Adventure"
    >
      Adventure
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Action"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Action"
    >
      Action
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Crime"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Crime"
    >
      Crime
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Comedy"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Comedy"
    >
      Comedy
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Drama"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Drama"
    >
      Drama
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Fantasy"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Fantasy"
    >
      Fantasy
    </Link>
  </li>
  <li
    className="catalog__genres-item "
    key="Thriller"
  >
    <Link
      className="catalog__genres-link"
      onClick={[Function]}
      to="/?filter=Thriller"
    >
      Thriller
    </Link>
  </li>
</ul>
`;
